CONTRIBUTE.md

```diff
@@ -0,0 +1,333 @@
+# JavaScript Interactive Examples Contributing Guide
+
+A guide for people wanting to contribute interactive JavaScript examples.
+
+## Naming Things
+
+It probably makes sense to name things in a consistent way throughout all examples.  Where possible the naming should ‘reflect’ the type of that object or variable.
+
+### Numerical Suffixes
+
+One way we could maintain a certain consistency is to use the type of the object with a numerical suffix.
+
+For example:
+
+```
+object1 = {};
+var string1 = “a string”;
+var array1 = ['a', 'b', 'c'];
+var array2 = [1, 2, 3];
+```
+
+Note it is suggested that even if there is only one instance of that object then it should be suffixed with a `1`.
+
+Note also although we could use the same name for multiple instances, it's suggested that we use a progressive numbering system to minimise confusion should the reader want to experiment with the example code.
+
+For example:
+
+```
+var array1 = ['a', 'b', 'c'];
+
+console.log(array1);
+// expected output: Array ["a", "b", "c"]
+
+var array2 = [1, 2, 3];
+
+console.log(array2);
+// expected output: Array [1, 2, 3]
+```
+
+An alternative to using numerical suffixes could be to use more descriptive suffixes, such as `arrayLetters` and `arrayNumbers` in the above examples.
+
+This may make more sense in the context of the following example:
+
+```
+var collatorDe = new Intl.Collator('de', { usage: 'search', sensitivity: 'base' });
+var collatorFr = new Intl.Collator('fr', { usage: 'search', sensitivity: 'base' });
+```
+
```

Comment:
Is it worth having a third section like "descriptive names". For example:

```
let fruit = ["apples", "bananas", "oranges"]
```

...this only works for less abstract examples. But less abstract example might be better anyway, as the "tangible examples" section says.

Reply:
Yes we could do - I'm torn about whether to be consistent or allowing the flexibility to be descriptive - and I'm fine with both, but thinking now I may have been leaning to strongly on the consistency side and so we may want to change back some naming of the array examples I recently PR'd (https://github.com/mdn/interactive-examples/pull/350)